bodhi/client/__init__.py

```diff
@@ -337,6 +331,17 @@ def download(url, **kwargs):
                         click.echo("WARNING: download of {0} failed!".format(build['nvr']))
 
 
+def read_notes(**kwargs):
+    if kwargs['notes_file'] is not None:
+        if kwargs['notes'] is None:
+            with open(kwargs['notes_file'], 'r') as fin:
+                kwargs['notes'] = fin.read()
+        else:
+            click.echo("ERROR: Cannot specify --notes and --notes-file")
+            sys.exit(1)
+    return kwargs['notes']
```

Comment:
There's sort of an unspoken (or maybe it is spoken somewhere deep in Python's docs…) convention in Python that I think is a good rule to live by - functions/methods that change an object should not return the object, and functions/methods that do not change an object should return the result. If you study Python's built-in object APIs, they all follow this convention. For example, ```list.append``` does not return the list after modifying it:

```
>>> a = [1, 2, 3]
>>> x = a.append(4)
>>> print x
None
```

As another example, ```random.choice()``` does return an object and does not modify the list, where ```random.shuffle()``` does modify the object and thus does not return it:

```
>>> import random
>>> a = [1, 2, 3]
>>> x = random.choice(a)
>>> x
3
>>> a
[1, 2, 3]
>>> x = random.shuffle(a)
>>> print x
None
>>> a
[3, 1, 2]
```

It's actually a pretty nice convention, because when software follows it you can pretty easily guess whether the function is modifying objects or not without trawling around the docs too much (even though some people like @jeremycline read docs instead of eating food).

All this to say, you can design this function to work either way (modify the notes in place, or simply return what the notes should be), but it would be conventional to not have it do both. If I were to decide, I think it'd be simplest to just have it modify the dict in place and just drop the return statement. Of course, you shoudl also document this behavior in the function's docblock.